bundles/org.openhab.core.io.net/pom.xml

```diff
@@ -20,6 +20,11 @@
       <artifactId>org.openhab.core</artifactId>
       <version>${project.version}</version>
     </dependency>
+    <dependency>
+      <groupId>org.openhab.core.bundles</groupId>
+      <artifactId>org.openhab.core.thing</artifactId>
+      <version>${project.version}</version>
+    </dependency>
```

Comment:
It might be better to just add a utility class in `org.openhab.core.thing` to generate these names. That will still allow for adding a dependency on`org.openhab.core.io.net` in `org.openhab.core.thing` one day. Which would cause a dependency cycle when `org.openhab.core.io.net` has a depenency on `org.openhab.core.thing`

Reply:
Ok

Reply:
Maybe it is also possible to use the `AbstractUID` type instead of `ThingUID`? Then the logic would be more generic and there would also be no need for the dependency.

Reply:
`AbstractUID` will not give me access to the individual segments.

Reply:
Done. Utility method moved inside package org.openhab.core.thing.util